src/content/blog/node-test-migration.mdx

```diff
@@ -0,0 +1,131 @@
+- two fewer dependencies to install and maintain in our monorepo: `mocha` and `chai`;
+- maintainability, there are way more people involved in the Node.js project (hence the Node.js test runner);
+- we believe that the test runner will get better and better with the time, and eventually save some time in our CI workflows.
+
+## From an idea to a PoC
+
+The Astro monorepo has more than 500 testing suites: between integration tests and unit tests, we have 664 suites, with a total of 1603 tests. Tha majority of these tests are **integration** tests.
+
+An integration test, in our monorepo, means the creation of tiny Astro project, building a project with a specific environment - development, static generation (SSG) or dynamic generation (SSR), and run assertions over the **built** pages. That's right, each integration test requires `vite` to build and bundle the project.
+
+We didn't start the migration right away. Before taking a final decision, we wanted to make sure that moving away from Mocha was a mistake. Despite its quirks, Mocha is a **fine** test runner, it's been around for a long time, it is battle-tested. If you use Mocha, you are in good hands.
+
+The idea of the PoC was to understand:
+- the flexibility of the Node.js CLI arguments and how customisable can be the test reporters;
+- the speed of execution of the testing suites;
+- the overall developer experience;
+
+### How we started
+
+We started by migrating only **one** of our packages that didn't use `astro`'s integration suite, `create-astro`. This was a good opportunity to play with the built-in assertion library `node:assert`, to learn about the options, and its performance compared to Mocha.
+
+Since `create-astro` only had a handful of tests, it was relatively easy to migrate the test files to use `node:test` and `node:assert`, instead of `mocha` and `chai`. After that, the only thing left is to update the `mocha` command to `node --test` to execute the tests. However, we quickly ran into issues using the `node --test` command, including:
+
+- It had trouble parsing the glob syntax when passing multiple arguments, e.g. `node --test "test/*.test.js" --test-name-pattern="foo"`.
+- It wasn't possible to pass the `--test-concurrency` flag (only available in Node.js 21 and above), but can be worked around by using the [programmatic API](https://nodejs.org/api/test.html#testname-options-fn) `concurrency` option.
+- Nitpicking, but the argument names were verbose, e.g. `--test-name-pattern` instead of `--match, -m` arguments, `--test-timeout` instead of `--timeout, -t` arguments, etc.
+
+Hence, to solve these issues, we created a [custom script](https://github.com/withastro/astro/blob/3757a212c01cee03b8e96e1e9697a187ae7cc290/scripts/cmd/test.js) which can be called with the `astro-scripts test` command. This decision also proved to be useful to enable more workarounds as we'll see later.
+
+### The pandora box
```

Comment:
```suggestion
### Opening Pandora's box
```
At the very least, Pandora is a name and should be capitalized. I think in English we *always* hear this phrase as the possessive "Pandora's box", so it sounded odd to hear it not that way to me. So, here's one suggestion of a way to use the phrase that way.